resources/lib/functions.py

```diff
@@ -134,6 +134,25 @@ def mainEntryPoint():
         checkService()
         checkServer(notify=False)
         showContent(sys.argv[0], int(sys.argv[1]), params)
+    elif mode == "SEARCH":
+        # plugin://plugin.video.embycon?mode=SEARCH
+        checkService()
+        checkServer(notify=False)
+        xbmcplugin.setContent(int(sys.argv[1]), 'files')
+        showSearch()
+    elif mode == "NEW_SEARCH":
+        # plugin://plugin.video.embycon?mode=NEW_SEARCH&item_type=<Movie|Series|Episode>
+        if 'SEARCH_RESULTS' not in xbmc.getInfoLabel('Container.FolderPath'):  # don't ask for input on '..'
```

Comment:
not sure this is the best way, if a user hits back to "fix" their search or do another search then they are taken back to the root and have to navigate back to the search area they are interested in again.
i.e.
I do a search for super
lots of result and I want to go back and refine the search to superman
when I hit back a go back to the root and skip the search input

Perhaps there is a better way.